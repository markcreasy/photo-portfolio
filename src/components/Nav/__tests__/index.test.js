import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
  //baseline test
  it('renders', () => {
    render(<Nav />);
  })
  //snapshot test
  it('Matches snapshot', () => {
    const {asFragment} = render(<Nav />)
    expect(asFragment()).toMatchSnapshot();
  })
})

describe('Emoji is visible', () => {
  it('inserts emoji into h2', () => {
    const {getByLabelText} = render(<Nav />);
      expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
  })
})

describe('Links are visible', () => {

  it('Inserts text into links', () => {
    const {getByTestId} = render(<Nav />)
    expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
    expect(getByTestId('about')).toHaveTextContent('About me');
  })
})
