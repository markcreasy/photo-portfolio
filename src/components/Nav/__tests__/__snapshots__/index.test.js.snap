// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component Matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
           ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span
            data-testid="contact"
          >
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span
            data-testid="commercial"
          >
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span
            data-testid="portraits"
          >
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span
            data-testid="food"
          >
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span
            data-testid="landscape"
          >
            landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
